src/evidently/EvidentlyManager.ts

```diff
@@ -0,0 +1,261 @@
+import { Dispatch } from '../dispatch/Dispatch';
+import { Config } from '../orchestration/Orchestration';
+import { EventCache } from '../event-cache/EventCache';
+import { NIL_UUID } from '../sessions/SessionManager';
+import {
+    BatchEvaluateFeatureRequest,
+    BatchEvaluateFeatureResult,
+    ContextType,
+    EvaluationResult,
+    EvaluationResults,
+    InitializeFeaturesRequest
+} from './types';
+import { Response } from 'node-fetch';
+
+const MAX_EVIDENTLY_FEATURES_PER_EVENT = 10;
+
+const REQUEST_CACHE_KEY = 'lastRequest';
+const EVALS_CACHE_KEY = 'cachedEvaluations';
+
+export class EvidentlyManager {
+    private eventCache: EventCache;
+    private dispatchManager: Dispatch;
+    private config: Config;
+    private isEnabled: boolean;
+
+    private lastRequest?: InitializeFeaturesRequest;
+    private cachedEvaluations: EvaluationResults = {};
+    private lastPromise: Promise<EvaluationResults> = Promise.resolve({});
+    private cacheValidCheck = 0;
+
+    constructor(
+        config: Config,
+        eventCache: EventCache,
+        dispatchManager: Dispatch
+    ) {
+        this.eventCache = eventCache;
+        this.dispatchManager = dispatchManager;
+        this.config = config;
+        this.isEnabled = config.evidentlyConfig.project !== undefined;
+    }
+
+    public initializeFeatures(request: InitializeFeaturesRequest) {
+        if (!this.isEnabled) {
+            throw new Error('Evidently is not enabled');
+        } else if (request.features.length > MAX_EVIDENTLY_FEATURES_PER_EVENT) {
+            throw new Error(
+                `Can only request up to ${MAX_EVIDENTLY_FEATURES_PER_EVENT} features at a time`
```

Comment:
I don't think this is a blocker for this release, but it should be a follow up. I think we need to add the exception type to the `name` parameter of the error object for customers to do handle errors properly: https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/

In the case `error.name = 'ValidationException'`

List of recommended exceptions:
https://w.amazon.com/bin/view/AWS/API_Standards/Exceptions/#HRecommendedExceptions

Reply:
So everywhere else in the web client code uses plain JS errors like this. I agree it'd be better to throw SDK compatible errors, but I think it may be out of scope right now to change to that because the web client should be self consistent. @qhanam What do you think?